pkg/kubelet/cri/remote/remote_image.go

```diff
@@ -82,7 +82,7 @@ func NewRemoteImageService(endpoint string, connectionTimeout time.Duration, tp
 
 	service := &remoteImageService{timeout: connectionTimeout}
 
-	if err := service.determineAPIVersion(conn, endpoint); err != nil {
+	if err := service.determineAPIVersion(ctx, conn, endpoint); err != nil {
```

Comment:
@liggitt 👀  , we are cascading a context with a timeout that is supposed to be used only for the dialer, based on my reading of the code

Reply:
it doesn't seem the way the code is structured, it has an specific option to set the timeout, cascading the other timeout doesn't seem the intention

Reply:
Note that r.timeout is set to connectionTimeout.  That means both WithTimeout calls are basically identical.

The duplicate timeout setting can be removed, but I don't think it is the cause of the failures.

Reply:
but is that on purpose?
despite the result is correct I think we should move to a better readability of the code, for the future us,  do you think we´ll understand why those contexts are plumbed?

Reply:
I'll fix the duplicate timeout when I re-try this PR.

Reply:
we can remove the one in determineAPIVersion

Reply:
your call, I really didn't dig into it, but at first sight was confusing to me , maybe is to others
Thanks